pkg/config/manifest.go

```diff
@@ -723,16 +744,30 @@ func (m *Manifest) resolveValue(key string) (interface{}, error) {
 						"unknown credential definition, no environment variable or path specified",
 					)
 				}
+
+				// if replacement has been set, add to list of sensitive values
+				if replacement != nil {
+					m.sensitiveValues = append(m.sensitiveValues, reflect.ValueOf(replacement).String())
+				}
 			}
 		}
 	case "outputs":
 		if o, exists := m.outputs[sourceName]; exists {
 			replacement = o
+			// Porter considers all outputs as sensitive
+			if replacement != nil {
+				m.sensitiveValues = append(m.sensitiveValues, reflect.ValueOf(replacement).String())
+			}
 		}
 	case "dependencies":
 		for _, dep := range m.Dependencies {
 			if dep.Name == sourceName {
-				return dep.resolveValue(key)
+				replacement, err := dep.resolveValue(key)
+				// Retrieve updated list of sensitive values from dependency and add to our list
```

Comment:
I think I see why you put this here, but it really will need to be refactored or improved later. It's problematic that we need to remember to do this after calling `dep.resolveValue` and points to a leaky abstraction with how we are either managing our dependencies or how we are collecting and managing these sensitive values I think.